spec/controllers/partners_controller_spec.rb

```diff
@@ -70,4 +70,20 @@
       expect(subject).to redirect_to(partners_path)
     end
   end
+  describe "DiaperPartnerClient first" do
+    let(:partner) { create(:partner) }
+    subject { put :approve_application, params: { organization_id: partner.organization.id, 
+      id: partner.id }
+    }
+    it "Always redirects to partners_path" do
+      expect(subject).to redirect_to partners_path
+    end
+    context "Working Partner API" do
+      it "Changes Status to Approved" do
+        allow(DiaperPartnerClient).to receive(:put){ Net::HTTPSuccess.new(1.0, '200', 'OK') }
```

Comment:
Nice!

Reply:
Thank you! I had help from Chase, 2 hours later and we finally ended up with working tests!